### DISCUSSION TITLE
Error while building kvorkcs ,it sames that downloading jemalloc failed

### DISCUSSION THREAD

**Discussion by sjpcg** [apache/kvrocks#712] [Q&A]
Error Detail  as below:
[ 11%] Creating directories for 'jemalloc-populate'
[ 22%] Performing download step (download, verify and extract) for 'jemalloc-populate'
-- Downloading...
   dst='/data0/kvrocks/incubator-kvrocks/build/_deps/jemalloc-subbuild/jemalloc-populate-prefix/src/12cd13cd418512d9e7596921ccdb62e25a103f87.zip'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/jemalloc/jemalloc/archive/12cd13cd418512d9e7596921ccdb62e25a103f87.zip'
CMake Error at jemalloc-subbuild/jemalloc-populate-prefix/src/jemalloc-populate-stamp/download-jemalloc-populate.cmake:170 (message):
  Each download failed!

    error: downloading 'https://github.com/jemalloc/jemalloc/archive/12cd13cd418512d9e7596921ccdb62e25a103f87.zip' failed
          status_code: 1
          status_string: "Unsupported protocol"
          log:
          --- LOG BEGIN ---
          Protocol "https" not supported or disabled in libcurl

  Closing connection -1

  

          --- LOG END ---
          
    

gmake[2]: *** [jemalloc-populate-prefix/src/jemalloc-populate-stamp/jemalloc-populate-download] Error 1
gmake[1]: *** [CMakeFiles/jemalloc-populate.dir/all] Error 2
gmake: *** [all] Error 2

CMake Error at build/cmake/share/cmake-3.23/Modules/FetchContent.cmake:1088 (message):
  Build step for jemalloc failed: 2
Call Stack (most recent call first):
  build/cmake/share/cmake-3.23/Modules/FetchContent.cmake:1217:EVAL:2 (__FetchContent_directPopulate)
  build/cmake/share/cmake-3.23/Modules/FetchContent.cmake:1217 (cmake_language)
  cmake/jemalloc.cmake:29 (FetchContent_Populate)
  CMakeLists.txt:50 (include)

-- Configuring incomplete, errors occurred!
See also "/data0/kvrocks/incubator-kvrocks/build/CMakeFiles/CMakeOutput.log".

**Comment by git-hulk** [comment]
It looks that libcurl in `cmake` command wasn't enabled. Does @PragmaTwice see this error before?

**Comment by git-hulk** [comment]
Maybe it's caused by this PR: https://github.com/apache/incubator-kvrocks/pull/583

**Comment by PragmaTwice** [comment]
Could you try to download cmake binary directly from [here](https://github.com/Kitware/CMake/releases) instead of building from its source code?

**Reply by sjpcg** [reply]
OK，i'll try

**Reply by sjpcg** [reply]
Can you give me one download  link of cmake binary for the right version and format ? Thank you

**Reply by tisonkun** [reply]
@sjpcg If you are on macOS, you can install cmake with [Homebrew](https://brew.sh/) by `brew install cmake`.

Otherwise you can find the latest version at https://cmake.org/download/ and that should work. Please try to resolve such general questions with Google and StackOverflow first instead of asking for a step by step instructions.

**Reply by sjpcg** [reply]
ok ,thx
